daemon/images/image_pull.go

```diff
@@ -73,6 +73,9 @@ func (i *ImageService) PullImage(ctx context.Context, image, tag string, platfor
 			progress.Messagef(po, "", `WARNING: %s`, err.Error())
 			logrus.WithError(err).WithField("image", image).Warn("ignoring platform mismatch on single-arch image")
 		}
+		if err != nil && img == nil {
```

Comment:
What are the other special cases where `err != nil && img != nil`? Why is silently ignoring the error the correct behaviour?

Reply:
See the comment above:
```
// Note that this is a special case where GetImage returns both an image
// and an error: https://github.com/docker/docker/blob/v20.10.7/daemon/images/image.go#L175-L183
```

Reply:
Right, fixed